### PR TITLE
Add an example with Buttons to self-assign roles

### PR DESCRIPTION
**Pull request by SlimShadyIAm**
## Summary

This is a simple example of how to create a persistent view with Buttons for self-assignable roles. Similar to previous approaches with reaction roles.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting, ...)

### CODE REVIEW

**Comment by Lulalaby** [comment]
Cool thing. Gonna take a look later

File: examples/views/button_roles.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+"""
+Let users assign themselves roles by clicking on Buttons
+Make sure to add the cog to the initial_extensions list
+in main.py
+"""
+
+# I chose to have emojis in my buttons 
+# you could also do this as just a list of role IDs
+emoji_to_role_id = {
+    "<:gircoin:846511056137748571>": "846383888003760217",
```
Comment (Lulalaby) [inline examples/views/button_roles.py:14]: I think that is bad practice

File: examples/views/button_roles.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        guild = self.bot.get_guild(123)
+        for emoji, role_id in emoji_to_role_id.items():
+            role = guild.get_role(int(role_id))
+            view.add_item(RoleButton(role, emoji))
+
+        # add the view to the bot so it will watch for button interactions
+        self.bot.add_view(view)
+
+def setup(bot):
+    bot.add_cog(ButtonRoleCog(bot))
```
Comment (Lulalaby) [inline examples/views/button_roles.py:66]: You could document more

**Review by Lulalaby — CHANGES_REQUESTED** [review]

**Comment by SlimShadyIAm** [comment]
I've simplified the example by removing emojis from the buttons and also added more documentation.

**Review by Lulalaby — APPROVED** [review]

**Comment by BobDotCom** [comment]
It would

**Review by BobDotCom — APPROVED** [review]

**Pull request merged**